src/pretalx/submission/models/submission.py

```diff
@@ -180,20 +181,36 @@ def get_duration(self):
             return self.submission_type.default_duration
         return self.duration
 
-    def confirm(self, person=None, force=False, orga=False):
-        if self.state not in [SubmissionStates.ACCEPTED] and not force:
-            raise SubmissionError(_('Submission must be accepted, not {state} to be confirmed.').format(state=self.state))
+    def _set_state(self, new_state, force=False):
+        """
+        check if the new state is valid for this Submission (based on SubmissionStates.valid_next_states).
+        if yes, set it and save the object. if no, raise a SubmissionError with a helpful message.
+        """
+
+        valid_next_states = SubmissionStates.valid_next_states[self.state]
```

Comment:
Just to be really safe, I'd prefer a `.get(self.state, [])` here – if the db is corrupted, somebody sets a wrong state manually, or we drop a state without a data migration because we're lazy, our code will not break horribly, just intentionally.